api/auth.py

```
@@ -338,8 +338,30 @@ async def require_active_subscription_with_usage_tracking(
+    # Special handling for dashboard endpoints - allow burst usage
+    dashboard_endpoints = ["general_api", "data_summary", "jobs_list", "subscription_status"]
+    if endpoint in dashboard_endpoints:
+        # Get recent usage in last 5 minutes for burst detection
+        recent_period = datetime.now(timezone.utc) - timedelta(minutes=5)
+        recent_usage = _get_current_usage(user.id, usage_type, recent_period, db)
+        
+        # Allow up to 20 dashboard calls in 5 minutes (4 calls per minute)
+        if recent_usage >= 20:
+            raise HTTPException(
+                status_code=status.HTTP_429_TOO_MANY_REQUESTS,
+                detail=f"Dashboard rate limit exceeded. Please wait a moment before refreshing.",
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)